src/main/java/control/DetailViewController.java

```diff
@@ -497,55 +474,104 @@ private void activeBlockChangedDirector() {
             .setText(detailView.formatDouble(shotBlock.getPaddingAfter()));
-        activeBlock = shotBlock;
-        initInstrumentsDropdown(shotBlock);
+        activeDirectorBlock = shotBlock;
         detailView.setVisible();
         // Re-init the detail view with new data
         manager.getRootPane().getRootHeaderArea().reInitHeaderBar(detailView);
         this.reInitForDirectorBlock();
     }
 
     /**
-     * Initialize drop down menu.
+     * Initialize drop down menu for director camera selection.
      */
-    private void initDropDown() {
+    private void initDirectorCameraDropdown() {
         StyledMenuButton cameraButtons = ((DirectorDetailView) detailView).getSelectCamerasButton();
         cameraButtons.setBorderColor(TweakingHelper.getColor(0));
         cameraButtons.setFillColor(TweakingHelper.getBackgroundColor());
-        activeBlockBoxes = new ArrayList<>();
+        activeCameraBoxes = new ArrayList<>();
+
+        cameraButtons.showingProperty().addListener(createCameraDropdownListener(cameraButtons));
+    }
+
+    /**
+     * Initialize drop down menu for camera instrument selection.
+     */
+    private void initCameraInstrumentDropdown() {
+        StyledMenuButton instrumentsButtons =
+                ((DetailView) detailView).getSelectInstrumentsButton();
+        instrumentsButtons.setBorderColor(TweakingHelper.getColor(0));
+        instrumentsButtons.setFillColor(TweakingHelper.getBackgroundColor());
+        activeInstrumentBoxes = new ArrayList<>();
+
+        instrumentsButtons.showingProperty().addListener(
+                createInstrumentsDropdownListener(instrumentsButtons));
+    }
+
+    /**
+     * Creates ChangeListener for the Instruments Dropdown checkboxes.
+     * @param buttons the dropdown with checkboxes.
+     * @return the ChangeListener.
+     */
+    private ChangeListener<Boolean> createInstrumentsDropdownListener(StyledMenuButton buttons) {
+        return new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable,
+                                Boolean oldValue, Boolean newValue) {
+                if (newValue) {
+                    List<Instrument> instruments = activeCameraBlock.getInstruments();
+
+                    for (int i = 0; i < manager.getScriptingProject().getInstruments().size();
+                         i++) {
+                        Instrument instrument = manager.getScriptingProject()
+                                .getInstruments().get(i);
+                        StyledCheckbox checkbox = new StyledCheckbox(instrument.getName(),
+                                instruments.contains(instrument));
+                        activeInstrumentBoxes.add(checkbox);
+                        CustomMenuItem item = new CustomMenuItem(checkbox);
+                        item.setHideOnClick(false);
+                        buttons.getItems().add(item);
+
+                        int j = i;
```

Comment:
Was a scoping issue, but I fixed that and forgot to remove it. Consider it gone.